src/utils/github.py

```diff
@@ -61,20 +65,28 @@ def create_headers(bearer: str) -> dict:
             return headers
 
         if self.gh_app_id and self.gh_pem_path:
-            response = requests.get(
-                "https://api.github.com/app/installations",
-                headers=create_headers(self.gh_jwt),
-            )
-            gh_tokens_url = response.json()[0]["access_tokens_url"]
-
-            response = requests.post(
-                gh_tokens_url, headers=create_headers(self.gh_jwt)
-            )
-            gh_token = response.json()["token"]
-
-            data = {"ref": "master"}
-            response = requests.post(
-                f"{repository}/actions/workflows/{workflow}/dispatches",
-                headers=create_headers(gh_token),
-                json=data,
-            )
+            try:
+                response = requests.get(
+                    "https://api.github.com/app/installations",
+                    headers=create_headers(self.gh_jwt),
+                )
+                response.raise_for_status()
+                gh_tokens_url = response.json()[0]["access_tokens_url"]
+
+                response = requests.post(
+                    gh_tokens_url, headers=create_headers(self.gh_jwt)
+                )
+                response.raise_for_status()
+                gh_token = response.json()["token"]
+
+                data = {"ref": "master"}
+                response = requests.post(
+                    f"{repository}/actions/workflows/{workflow}/dispatches",
+                    headers=create_headers(gh_token),
+                    json=data,
+                )
+                response.raise_for_status()
+                self.logger.info(f"GH workflow triggered: {workflow}")
+
+            except Exception as e:
+                self.logger.error(f"GH workflow run failed: {e}")
```

Comment:
**[Nitpick, non-blocking]** Is there a good reason to use an acronym here? I think using the full service name will make this log line more search-friendly:

```suggestion
                self.logger.error(f"GitHub workflow run failed: {e}")
```